### PR TITLE
Added v2 changes in migration docs

### PR DESCRIPTION
**Pull request by marcellorinaldo**
Signed-off-by: Marcello Martina <martina.marcello.rinaldo@outlook.com>

My first PR to this project. Here I added the major changes that I have found while migrating from 1.4.199 to 2.0.206. Feel free to comment on it and to propose other changes.

### CODE REVIEW

**Comment by andreitokar** [comment]
Thank You for this contribution.
Unfortunately, as of now, **h2database.github.io** repository is only used to hold web site content, which is generated as part of the H2 release, from templates in **h2database** repository. Can you please move your changes to create a PR against this file: https://github.com/h2database/h2database/blob/master/h2/src/docsrc/html/migration-to-v2.html ?

File: html/migration-to-v2.html [hunk 1]
```diff
@@ -5,185 +5,251 @@
...
+                    <p>
+                        Migrating an old database to the new version works most of the times. However, there are a
+                        couple of important changes in the new version to keep in mind:
+                    </p>
+                    <ul>
+                        <li>
+                            <b>Restricted maximum allowed length limits:</b> there are new length limits for for
+                            non-LOB data types. For example, for the VARBINARY type the maximum allowed size is now
+                            1048576 (1MB)
+                            instead of the previous 16777216 (16MB). Adjust the tables before migration.
```
Comment (katzyn) [inline html/migration-to-v2.html:205]: Previous declared limit was 2^31-1, but it wasn't possible to create a value with this length. I have no idea where you found 16MB, but this value is incorrect.

File: html/migration-to-v2.html [hunk 2]
```diff
@@ -5,185 +5,251 @@
...
+                        </li>
+                        <li>
+                            <b>VARCHAR length changes unit syntax:</b> The length of the VARCHAR datatype cannot
+                            be more specified using CHAR but CHARACTERS or OCTETS.
```
Comment (katzyn) [inline html/migration-to-v2.html:209]: H2 never supported Oracle-style units officially, I think people who use undocumented features are on their own and this information isn't that important to be listed here.

File: html/migration-to-v2.html [hunk 3]
```diff
@@ -5,185 +5,251 @@
...
+                        </li>
+                        <li>
+                            <b>IDENTITY syntax changed when type is specified:</b> if the type for IDENTITY is
+                            specified, then the clause needs to be expanded as INTEGER GENERATED ALWAYS AS IDENTITY.
+                            Using just INTEGER IDENTITY is no more working.
+                        </li>
+                        <li>
+                            <b>Usage of statement.setTimestamp() is discouraged:</b> use setObject(parameterIndex,
+                            value) with java.time.LocalDateTime parameter instead. For example: stmt.setTimestamp(2,
+                            timestamp, this.utcCalendar); becomes stmt.setObject(2, timestamp.toLocalDateTime());.
```
Comment (katzyn) [inline html/migration-to-v2.html:219]: Where did you find that information?

Documentation of 1.4.200 already recommended JSR-310 data types instead, there is nothing new it that area in H2 2.0, only some bugs were fixed.

And documentation recommends usage of new data types. I doesn't recommend to convert legacy data types to modern ones for H2. If you still want to use defective by design legacy types, you can pass them to H2 as is.

File: html/migration-to-v2.html [hunk 4]
```diff
@@ -5,185 +5,251 @@
...
+                        </li>
+                        <li>
+                            <b>IDENTITY() function is removed:</b> statements like CALL IDENTITY(); aren't working
+                            anymore, use another method like SELECT MAX(ID) FROM test;.
```
Comment (katzyn) [inline html/migration-to-v2.html:223]: Even in H2 1.* `IDENTITY()` is a wrong function, `SCOPE_IDENTITY()` was far better, at least it cannot return unrelated value from a trigger. But both these functions were removed.

`SELECT MAX(ID)` as also a wrong solution. The proper solution is to use data change delta tables or to use `Statement.getGeneratedKeys()`.

File: html/migration-to-v2.html [hunk 5]
```diff
@@ -5,185 +5,251 @@
...
+                        </li>
+                        <li>
+                            <b>LOG connection setting removed:</b> the "LOG=0" setting at the end of the URL (like
+                            "jdbc:h2:file:/tmp/test;LOG=0") that disables the transaction log on persisted databases has
+                            been removed.
```
Comment (katzyn) [inline html/migration-to-v2.html:228]: This setting was for PageStore, this backend was removed from H2.

**Review by katzyn — COMMENTED** [review]
Please, don't change the formatting of this file in the new pull request in the proper repository and follow the same formatting as used in our documentation everywhere.

**Pull request closed without merging**